src/edu/ucsb/cs56/projects/games/subsink/Screen.java

```diff
@@ -0,0 +1,212 @@
+
+            MouseAdapter ma = new MouseAdapter() {
+
+                @Override
+                public void mouseClicked(MouseEvent e) {
+                    String newItem = null;
+                    for (String text : menuItems) {
+                        Rectangle bounds = menuBounds.get(text);
+                        if (bounds.contains(e.getPoint())) {
+                            newItem = text;
+                            break;
+                        }
+                    }
+                    if (newItem != null && !newItem.equals(selectedMenuItem)) {
+                        selectedMenuItem = newItem;
+                        repaint();
+                    }
+                }
+
+                @Override
+                public void mouseMoved(MouseEvent e) {
+                    focusedItem = null;
+                    for (String text : menuItems) {
+                        Rectangle bounds = menuBounds.get(text);
+                        if (bounds.contains(e.getPoint())) {
+                            focusedItem = text;
+                            repaint();
+                            break;
+                        }
+                    }
+                }
+
+            };
+
+            addMouseListener(ma);
+            addMouseMotionListener(ma);
+
+
+        }
+
+        @Override
+        public void invalidate() {
+            menuBounds = null;
+            super.invalidate();
+        }
+
+        @Override
+        public Dimension getPreferredSize() {
+            return new Dimension(620, 480);
+        }
+
+
+        @Override
+        protected void paintComponent(Graphics g) {
+            super.paintComponent(g);
+            Graphics2D g2d = (Graphics2D) g.create();
+            if (menuBounds == null) {
+                menuBounds = new HashMap<>(menuItems.size());
+                int width = 0;
+                int height = 0;
+                for (String text : menuItems) {
+                    Dimension dim = painter.getPreferredSize(g2d, text);
+                    width = Math.max(width, dim.width);
+                    height = Math.max(height, dim.height);
+                }
+
+                int x = (getWidth() - (width + 10)) / 2;
+
+                int totalHeight = (height + 10) * menuItems.size();
+                totalHeight += 5 * (menuItems.size() - 1);
+
+                int y = (getHeight() - totalHeight) / 2;
+
+                for (String text : menuItems) {
+                    menuBounds.put(text, new Rectangle(x, y, width + 10, height + 10));
+                    y += height + 10 + 5;
+                }
+
+            }
+            for (String text : menuItems) {
+                Rectangle bounds = menuBounds.get(text);
+                boolean isSelected = text.equals(selectedMenuItem);
+                boolean isFocused = text.equals(focusedItem);
+                if (selectedMenuItem.equals("Start Game") || selectedMenuItem.equals("Try Again")) {
```

Comment:
This behavior-based code should _absolutely not_ be part of the paint routine. It should be part of the callback that actually handles the click.